File: Algorithm/344_reverseString/main.cpp
#include <bits/stdc++.h>
using namespace std;
void reverseString(vector<char> &s)
{
    int l = 0;
    int r = s.size() - 1;
    while (l < r)
    {
        int temp = s[l];
        s[l] = s[r];
        s[r] = temp;
        r--;
        l++;
    }
}
int main()
{
    vector<char> a = {'h', 'e', 'l', 'l', 'o'};
    reverseString(a);
    for (int i = 0; i < a.size(); i++)
    {
        cout << a[i];
    }
    cout << endl;
    return 0;
}

File: Algorithm/344_reverseString/main.py
from math import ceil

from pip import List


def reverseString(self, s: List[str]) -> None:
    size = len(s)
    mid = int(ceil(size/2))
    print(mid)
    for a in range(mid):
        temp = s[a]
        s[a] = s[size-1]
        s[size-1] = temp
        size -= 1


nums = ["h", "e", "l", "m",  "o"]
ans = reverseString(0, nums)
for n in nums:
    print(n)
